clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -258,7 +258,10 @@ public void poll(long now) {
         }
 
         pollHeartbeat(now);
-        maybeAutoCommitOffsetsAsync(now);
+        if (subscriptions.hasNoSubscription())
+            subscriptions.clearAssignment();
```

Comment:
I think I'm missing something. When calling `unsubscribe()` the method `onJoinPrepare` is not called (as it doesn't trigger a rebalance), so the assignment wouldn't get cleared. This would fail the added unit test `testSubscriptionChangesWithAutoCommitEnabled`. Could you please clarify?